packages/paste-core/components/ai-chat-log/src/useAIChatLogger.tsx

```diff
@@ -0,0 +1,42 @@
+import { uid } from "@twilio-paste/uid-library";
+import * as React from "react";
+
+import type { AIMessageVariants } from "./AIMessageContext";
+
+export type AIChat = {
+  id: string;
+  variant: AIMessageVariants;
+  content: React.ReactElement;
+};
+
+export type AIPartialIDChat = Omit<AIChat, "id"> & Partial<Pick<AIChat, "id">>;
+
+type PushAIChat = (chat: AIPartialIDChat) => void;
+type PopAIChat = (id?: string) => void;
+
+export type UseAIChatLogger = (...initialChats: AIPartialIDChat[]) => {
+  aiChats: AIChat[];
+  push: PushAIChat;
+  pop: PopAIChat;
+  clear: () => void;
+};
```

Comment:
I'm wondering about this API. Do you know why Shadi decided on this?

1. It seems that what I would otherwise call a message is considered an "Ai Chat" entity here. Usually I would think Chat is equal to Conversation, so one Chat would contain of many Messages.

So I would usually think messages would be more appropriate prop here. Also what OpenAI uses, some other LLMs for their API.

The second thing, I'm not sure if it's good to have initialChats as a parameter that alone is a spread array.

It would narrow down anything you could pass there in the future. I would suggest having an object to pass as config where you could have initialMessages prop. Like react-hook-form has initialValues.

Reply:
These are both good points. The reason we went with these decisions was solely to mirror the existing ChatLog API. Because many consumers are already familiar with that API, we wanted to replicate it as closely as possible with this one.